Fix feedbacks

forhadmethun opened the pull request:
- Segregate Request and Response messages
- Change `type` var to `messageType`
- IO for side effects (Random, UUID) in Orb.scala, Player.scala
- Use Vector instead of List
- Use evalTap instead of evalMap
- Remove unnecessary codec
- Use log4cats for logging
- Functional methods for increasing score / orbsAbsorbed count

forhadmethun committed:
Fix feedbacks
- Segregate Request and Response messages
- Change `type` var to `messageType`
- IO for side effects (Random, UUID) in Orb.scala, Player.scala
- Use Vector instead of List
- Use evalTap instead of evalMap
- Remove unnecessary codec
- Use log4cats for logging
- Functional methods for increasing score / orbsAbsorbed count

grzegorz-bielski commented on src/main/scala/services/GameService.scala:
```diff
@@ -12,24 +12,22 @@ import io.circe.parser.*
-  def publish(m: GameMessage)(using MonadThrow[F]): F[Unit] =
-    game.flatMap(_.publish(m))
-
+  def publish(msg: F[GameMessage])(using MonadThrow[F]): F[Unit] = {
```
Try passing `msg: GameMessage` instead. The upstream should be somehow flatMapped

forhadmethun committed:
Fix feedbacks
- Fix `publish` method og GameService to take GameMessage instead of F[GameMessage]
